### PR TITLE
Minor changes to get shell working in Cygwin

### PR DESCRIPTION
**Pull request by OllieTerrance**
As discussed in #514, this seems to be the minimum required to get a working prompt.

### CODE REVIEW

**Comment by scopatz** [comment]
Thanks a million for putting this in @OllieTerrance!

File: xonsh/foreign_shells.py [hunk 1]
```diff
@@ -243,9 +244,10 @@ def foreign_shell_data(shell, interactive=True, login=False, envcmd=None,
     elif currenv is not None:
         currenv = dict(currenv)
     try:
+        # start new session to avoid hangs (doesn't work on Cygwin though)
+        newsess = not (sys.platform == "cygwin")
```
Comment (scopatz) [inline xonsh/foreign_shells.py]: Would you mind making an `ON_CYGWIN` variable in `xonsh.platform`, please, importing and and using this instead? I think it is a little more general.

Comment (scopatz) [inline xonsh/foreign_shells.py]: I'd really appreciate it!

**Comment by scopatz** [comment]
Also, this could definitely use a CHANGELOG entry too, please.

**Comment by OllieTerrance** [comment]
Fair enough, this should hopefully cover it!  :smile:

**Comment by OllieTerrance** [comment]
The CI build seems to have stalled though, "Started Invalid date"...

**Comment by scopatz** [comment]
I think it is taking a bit to launch.  Travis seems slow today.

**Comment by scopatz** [comment]
Thanks a ton @OllieTerrance!

**Comment by GuillaumeLeclerc** [comment]
Should #514 be closed now ?

**Comment by OllieTerrance** [comment]
This was just a patch to get a shell prompt working.  The other ticket is more general -- I suspect there may be more issues on Cygwin than just job control...

**Pull request merged**